### PR TITLE
fix: IaC issue path

### PR DESCRIPTION
**Pull request by ofekatr**
### Description

- Fixes the types for the IaC issue path. Currently the path is represented as a slice of string tokens, when in reality tokens can also be integers.

### Checklist

- [X] Tests added and all succeed
- [X] Linted

### CODE REVIEW

File: application/server/lsp/message_types.go [hunk 1]
```diff
@@ -783,3 +785,18 @@ type IacIssueData struct {
 	Path          []string `json:"path"`
 	References    []string `json:"references,omitempty"`
 }
+
+type IacIssuePathToken struct {
+	Value any
+}
+
+func (c *IacIssuePathToken) String() string {
```
Comment (michelkaporin) [inline application/server/lsp/message_types.go:793]: @ofekatr Could you please move the conversion logic to the IaC scanner, when parsing IaC results and converting them to `snyk.Issue` representation? 

You should not be touching anything in `application/` or `domain/` at all.

File: application/server/lsp/message_types.go [hunk 2]
```diff
@@ -783,3 +785,18 @@ type IacIssueData struct {
...
+	switch val := c.Value.(type) {
+	case int:
+		return strconv.Itoa(val)
+	case string:
+		return val
+	default:
+		panic("invalid iac issue path component type")
```
Comment (michelkaporin) [inline application/server/lsp/message_types.go:800]: Please not panic and rely on logging or using `error_reporting.ErrorReporter` to [report errors](https://github.com/snyk/snyk-ls/blob/main/CONTRIBUTING.md#error-reporting).

**Review by michelkaporin — COMMENTED** [review]
Congratulations on your first PR in snyk-ls 🎉

File: application/server/lsp/message_types.go (hunk 1 above)
Comment (ofekatr) [reply application/server/lsp/message_types.go]: Fixed in https://github.com/snyk/snyk-ls/pull/269/commits/b866fde0c95531333e06c107927fe63999a87c71

File: application/server/lsp/message_types.go (hunk 2 above)
Comment (ofekatr) [reply application/server/lsp/message_types.go]: Fixed in https://github.com/snyk/snyk-ls/pull/269/commits/94151c1583e7f070a3985acd7224de1e3dba7ec2

File: .vscode/settings.json [hunk 3]
```diff
@@ -7,4 +7,23 @@
     "formatting.local": "github.com/snyk/snyk-ls"
   },
   "go.toolsManagement.autoUpdate": true,
+  "workbench.colorCustomizations": {
```
Comment (michelkaporin) [inline .vscode/settings.json:10]: We all have different tastes for colour, I wouldn't enforce this on the repository level 😉

File: application/server/lsp/message_types.go (hunk 1 above)
Comment (michelkaporin) [reply application/server/lsp/message_types.go]: Sorry, you've probably misunderstood what I meant. Could you have the logic in [this function](https://github.com/snyk/snyk-ls/blob/6d2a941c53e0effd4d5cd1c9e7f43143c506000f/infrastructure/iac/iac.go#L376)? It is part of IaC scanner and should belong there.

The changes outside of scanner shouldn't happen. Happy to have a quick call if you need little assistance 👍

Comment (ofekatr) [reply application/server/lsp/message_types.go]: Fixed in 8a2796a1b0f68622658e470ccf6b839ef2665088

File: .vscode/settings.json (hunk 3 above)
Comment (ofekatr) [reply .vscode/settings.json]: Added accidentally, fixed in f8652008d577e6580ea4351b03770cec65c91cec

File: infrastructure/iac/iac.go [hunk 4]
```diff
@@ -385,8 +409,56 @@ func (iac *Scanner) toAdditionalData(affectedFilePath string, issue iacIssue) sn
...
-		Path:          issue.Path,
+		Path:          iacIssuePath,
 		References:    issue.References,
+	}, nil
+}
+
+func (iac *Scanner) parseIacIssuePath(path []interface{}) ([]string, error) {
+	var pathTokens []string
+	for _, p := range path {
+		pt, err := newIacIssuePathToken(p, iac.errorReporter)
```
Comment (michelkaporin) [inline infrastructure/iac/iac.go:420]: You could avoid passing error reporter down to the `iacIssuePathToken` struct by capturing error using error reporter within this function.

File: infrastructure/iac/iac.go [hunk 5]
```diff
@@ -385,8 +409,56 @@ func (iac *Scanner) toAdditionalData(affectedFilePath string, issue iacIssue) sn
 		Issue:         issue.IacDescription.Issue,
 		Impact:        issue.IacDescription.Impact,
 		Resolve:       issue.IacDescription.Resolve,
-		Path:          issue.Path,
+		Path:          iacIssuePath,
 		References:    issue.References,
+	}, nil
+}
+
+func (iac *Scanner) parseIacIssuePath(path []interface{}) ([]string, error) {
```
Comment (michelkaporin) [inline infrastructure/iac/iac.go:417]: How about writing a few small unit tests for the function?

File: infrastructure/iac/iac.go [hunk 6]
```diff
@@ -324,7 +338,7 @@ func (iac *Scanner) getExtendedMessage(issue iacIssue) string {
 
 }
 
-func (iac *Scanner) toIssue(affectedFilePath string, issue iacIssue, fileContent string) snyk.Issue {
+func (iac *Scanner) toIssue(affectedFilePath string, issue iacIssue, fileContent string) (*snyk.Issue, error) {
```
Comment (bastiandoetsch) [inline infrastructure/iac/iac.go:341]: Why did you change it to a pointer return value? Any reason?

File: infrastructure/iac/iac.go (hunk 5 above)
Comment (bastiandoetsch) [reply infrastructure/iac/iac.go:417]: how about using `[]any` type? We are using go 1.19, and personally, I like it better than `[]interface{}`

File: infrastructure/iac/iac.go (hunk 4 above)
Comment (bastiandoetsch) [reply infrastructure/iac/iac.go:420]: Or just make it a caller function, e.g. func (iac ...) ...

**Comment by ofekatr** [comment]
@bastiandoetsch @michelkaporin Based on your feedback, I've decided to simplify the implementation. 
I'll let your team handle any potential refactors, should you see a need to.

File: infrastructure/iac/iac.go (hunk 6 above)
Comment (ofekatr) [reply infrastructure/iac/iac.go]: Changed in https://github.com/snyk/snyk-ls/pull/269/commits/d53881ee92568432ace2f313971cb3a49ed4c491?show-viewed-files=true&file-filters%5B%5D=

File: infrastructure/iac/iac.go (hunk 5 above)
Comment (ofekatr) [reply infrastructure/iac/iac.go]: Changed in https://github.com/snyk/snyk-ls/pull/269/commits/d53881ee92568432ace2f313971cb3a49ed4c491

File: infrastructure/iac/iac.go (hunk 4 above)
Comment (ofekatr) [reply infrastructure/iac/iac.go]: Changed in https://github.com/snyk/snyk-ls/pull/269/commits/d53881ee92568432ace2f313971cb3a49ed4c491

**Review by michelkaporin — APPROVED** [review]
That looks elegant now, let's merge it 👍

**Comment by ofekatr** [comment]
@michelkaporin It says my user is not authorized to merge the PR
